Linux/Car_Client/chronoscomm.cpp

```diff
@@ -49,6 +51,24 @@ ChronosComm::ChronosComm(QObject *parent) : QObject(parent)
             this, SLOT(tcpInputDisconnected()));
     connect(mTcpSocket, SIGNAL(error(QAbstractSocket::SocketError)),
             this, SLOT(tcpInputError(QAbstractSocket::SocketError)));
+
+	mLastHeabReceivedTimer.invalidate();
+	mLastHeabTimer = new QTimer(this);
+	connect(mLastHeabTimer, SIGNAL(timeout()), this, SLOT(checkLastHeabRestart()));
+	mLastHeabTimer->start(HEARTBEAT_TIME_MS);
+}
+
+void ChronosComm::checkLastHeabRestart(){
+	qDebug() << "Checking last heab received..";
+	if(mLastHeabReceivedTimer.isValid() && (mLastHeabReceivedTimer.elapsed() > HEARTBEAT_TIME_MS)){
+		qDebug()<< "HEAB timed out!";
+		emit heabTimeOut();
```

Comment:
I'm doing a (very!) similar solution for rcmm timeout. For rcmm it make sense to stop the timer here since the state will go from remote control to abort. What about heab? Is it by design that this timer will continue even when aborting?

Reply:
It is my understanding that HEAB is used to check if the connection is active or lost in some way. If you press ABORT in the GUC for example I think you should still handle HEAB to differentiate from the case when the HEAB connection is lost. (Even though in the current version booth cases lead to ABORT)

Reply:
It's worth double checking this part of the implementation. HEAB is an "active low" abort signal. In the same way if someone cuts the wire from a red abort button, cutting the connection should (via HEAB) cause an immediate abort same as if CCStatus became ABORT.

It is a bit unsafe that we use two methods for aborting - one via CCStatus and one via absence of HEAB - that way we won't be testing the two methods as much, and we risk making an implementation that only handles one.

Reply:
Booth methods use the same abort() function in the end. Having pressed abort in the GUC, this abort function is called. While in ABORTING heab is still beeing checked if present and in the case where no HEAB is received the same abort function is called again. There are no different abort functions, or am I missunderstanding your comment?